apps/ecommerce/frontend/src/types.ts

```diff
@@ -55,3 +56,18 @@ export interface ResourceCardProps {
   onMoveToTop?: Function;
   onMoveToBottom?: Function;
 }
+
+type ErrorBoundaryErrored = { hasError: true; error: Error; info: ErrorInfo };
+type ErrorBoundaryStandby = { hasError: false; error: null; info: null };
```

Comment:
Since these are types, is there a reason to set explicit values like `true/false/null` instead of `boolean` or `unknown`?

Reply:
Think of these types more as "states" of the component

Give a scenario where we are in an errored state, we can assume that `hasError` is true, this let's typescript know that error and info will also be defined. 

Since there's only two possible states this component could be in, this removes the need to if check both error/info or explicitly cast it as defined `eg. error!`